lib/stateTransition/validation/validateStateTransitionFeeFactory.js

```diff
@@ -0,0 +1,74 @@
+const ValidationResult = require('../../validation/ValidationResult');
+const stateTransitionTypes = require('../../stateTransition/stateTransitionTypes');
+const InvalidStateTransitionTypeError = require('../../errors/InvalidStateTransitionTypeError');
+const BalanceIsNotEnoughError = require('../../errors/BalanceIsNotEnoughError');
+
+const PRICE_PER_BYTE = 1;
+
+/**
+ * Validate state transition fee
+ *
+ * @param {DataProvider} dataProvider
+ * @param {function} parseTransactionOutPointBuffer
+ * @param {validateLockTransaction} validateLockTransaction
+ * @return {validateStateTransitionFee}
+ */
+function validateStateTransitionFeeFactory(
+  dataProvider,
+  parseTransactionOutPointBuffer,
+  validateLockTransaction,
+) {
+  /**
+   * @typedef validateStateTransitionFee
+   * @param {
+   * DataContractStateTransition|
+   * DocumentsStateTransition|
+   * IdentityCreateTransition} stateTransition
+   * @return {ValidationResult}
+   */
+  async function validateStateTransitionFee(stateTransition) {
+    const result = new ValidationResult();
+
+    const serializedStateTransition = stateTransition.serialize({ skipSignature: true });
+    const byteSize = Buffer.byteLength(serializedStateTransition);
+    const feeSize = byteSize * PRICE_PER_BYTE;
+
+    let balance;
+
+    if (stateTransition.getType() === stateTransitionTypes.IDENTITY_CREATE) {
```

Comment:
I believe, switch will look better here.

Reply:
But we can't declare variables inside switch, according to our eslint rules.